### ISSUE TITLE
Can't get analyzer server status for second analyzer context

### ISSUE THREAD

**Issue by skybrian** [dart-atom/dartlang#588]
While I had a Dart project open, i tried opening a second Dart project (in a separate Atom window) to try to reproduce an Analyzer bug. However, this didn't work.

it appears that if you choose Packages->Dart->Analyzer Service and click "View diagnostics", it shows the analyzer server for the first Dart project I opened. (In diagnostics, I only saw an Analyzer context for the first project.)

I tried closing the first project, but it didn't help. At this point, choosing "View diagnostics" shows ERR_CONNECTION_REFUSED.

Looking the output from 'ps ux' it appears the remaining Dart analyzer process was passed "--port=23072" at startup, but when I connect to this port on localhost, it's not there. But I still see Analyzer messages, somehow.

**Comment by devoncarew** [comment]
Yeah, we start one instance of the analysis server per atom window. If you have the 'start analysis server with diagnostics' setting on, we start the AS with the diagnostics flag. That's currently hard-coded to always use the same port (https://github.com/dart-atom/dartlang/blob/master/lib/analysis_server.dart#L39).

We could adjust the logic for that, or add a command to the AS to open a port for diagnostics and report back which port was opened.

**Comment by skybrian** [comment]
Okay, known bug then.

**Comment by skybrian** [comment]
For now, maybe disable the buttons for the second Atom window?

**Comment by devoncarew** [comment]
We should definitely improve the failure mode, if not making opening the diagnostics port more reliable.

**Comment by devoncarew** [comment]
I think what I'd really want here is an analysis server API call to open the diagnostics port, and return the port number to connect to. That way I don't have to start the server up with a specific port, and the server can bind to any port that's available.

**Comment by devoncarew** [comment]
blocked on https://github.com/dart-lang/sdk/issues/25395
